solidity/contracts/MezoAllocator.sol

```diff
@@ -225,20 +231,32 @@ contract MezoAllocator is IDispatcher, Ownable2StepUpgradeable {
     function withdraw(uint256 amount) external {
         if (msg.sender != address(stbtc)) revert CallerNotStbtc();
 
-        emit DepositWithdrawn(depositId, amount);
+        uint256 unallocatedBalance = tbtc.balanceOf(address(this));
+
+        if (amount > unallocatedBalance) {
+            uint256 amountToWithdraw = amount - unallocatedBalance;
+
+            emit DepositWithdrawn(depositId, amountToWithdraw);
+
+            if (amountToWithdraw < depositBalance) {
+                mezoPortal.withdrawPartially(
+                    address(tbtc),
+                    depositId,
+                    uint96(amountToWithdraw)
+                );
+            } else if (amountToWithdraw > depositBalance) {
+                revert WithdrawalAmountExceedsDepositBalance(
+                    amountToWithdraw,
+                    depositBalance
+                );
+            } else {
+                mezoPortal.withdraw(address(tbtc), depositId);
```

Comment:
Scenario:
`unallocatedBalance` = 0
`amount` = 10tBTC
`amountToWithdraw` == `depositBalance`
allocator calls `mezoPortal.withdraw` but there's a [fee in Mezo Portal](https://github.com/thesis/mezo-portal/blob/main/solidity/contracts/Portal.sol#L544). Say it's 0.1. So `MezoPortal` will return 9.9tBTC
Now, the allocator get's 9.9 which is less than 10. `tbtc.safeTransfer(address(stbtc), amount)` will revert in this case. Am I missing something? Is that known?

Reply:
The fee in MezoPortal is charged only for deposits with a minted receipt. For deposits made by the MezoAllocator, the withdrawal balance is expected to always match the deposit balance.
@pdyraga are there any plans to further change Mezo Portal withdrawals that would affect the MezoAllocator contract?

Reply:
There are no other fees and no other plans for fees in the Portal than the fee when someone minted the receipt token like stBTC. In this case, the fee is paid during the withdrawal. There are also fees from tBTC bridge like the redemption fee and minting fee but they do not affect Acre's MezoAllocator to my knowledge.